### PR TITLE
Issue #1949: Override h2s with h3s for multiselects for better doc structure - (BOB)

### LINKED ISSUE #1949: Django admin: make multi-selects comply to accessibility criteria
### Issue description

Make multi-selects comply to accessibility criteria. There is an accessibility issue on multi-selects (h2 should be an h3, or a <label> maybe)

See:
-  [slack thread](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1711644809098279) for screenshot
-  [slack thread](https://cisa-corp.slack.com/archives/C06K96473FA/p1711572361126209) for additional discussion

### Acceptance criteria

- [ ] ensure that multi-selects use h3 instead of h2, to be compliant with accessibility requirements.

### Additional context

Multi-selects are still used on User Model/Permissions.

Any accessibility audit that sees the structure:
```
<h2>Permissions</h2>
<h2>Available groups</h2>
<h2>Chosen groups</h2>
```
...
Will fail the page. It is structurally incorrect that Available groups and Chosen groups are sections on the level of Permissions on the User change_form 

### Links to other issues
Related to: #1950

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #1949

## Changes

- Dynamically remove h2s and replace with h3s
- Override CSS so that there are no visible changes on page
- Cleanup old multi list select JS code since we're no longer deploying those on DRs and DIs

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [x] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots

Screenshot 2024-04-26 at 12 27 56 PM

Screenshot 2024-04-26 at 12 31 57 PM

### CODE REVIEW

File: src/registrar/assets/js/get-gov-admin.js [hunk 1]
```diff
@@ -245,215 +243,56 @@ function checkToListThenInitWidget(toListId, attempts) {
...
-        var lastChild = relatedWidgetWrapper.lastChild;
-
-        // Check if lastChild is an element node (not a text node, comment, etc.)
-        if (lastChild.nodeType === 1) {
-            var previousSibling = lastChild.previousSibling;
-            // need to work around some white space which has been inserted into the dom
-            while (previousSibling.nodeType !== 1) {
-                previousSibling = previousSibling.previousSibling;
+// Replace h2 with more semantic h3
+function initializeWidgetOnList(list, parentId) {
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/assets/js/get-gov-admin.js:262]: Nope, this is all generated in JS on the DJANGO side (see my comment on top of this JS file), so my solution is to branch out of that.

File: src/registrar/assets/js/get-gov-admin.js [hunk 2]
```diff
@@ -245,215 +243,56 @@ function checkToListThenInitWidget(toListId, attempts) {
     let toList = document.getElementById(toListId);
     attempts++;
 
-    if (attempts < 6) {
-        if ((toList !== null)) {
+    if (attempts < 12) {
+        if ((toList && toList !== undefined)) {
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/assets/js/get-gov-admin.js:247]: Good one

File: src/registrar/assets/js/get-gov-admin.js [hunk 3]
```diff
@@ -245,215 +243,56 @@ function checkToListThenInitWidget(toListId, attempts) {
     let toList = document.getElementById(toListId);
     attempts++;
 
-    if (attempts < 6) {
-        if ((toList !== null)) {
+    if (attempts < 12) {
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/assets/js/get-gov-admin.js:246]: Magic number. Gotta wait for the Django JS to finish generating this stuff before we can manipulate it.

**Review by rileyorr — APPROVED** [review]
LGTM - great work! And thank you for helping me access the sandbox!

**Pull request merged**